src/store.c

```diff
@@ -339,47 +407,104 @@ static void writer_close(
     close(store->fd);
 }
 
+static struct rill_pairs* flip_values(struct rill_pairs* pairs)
+{
+    struct rill_pairs* inverted = rill_pairs_new(pairs->cap);
```

Comment:
I believe you should check for a `NULL` value here.

Reply:
Fixed - this function now does an in place swap, so no new allocs.